lib/pagseguro/subscription/subscription_search.rb

```diff
@@ -0,0 +1,135 @@
+module PagSeguro
+  class SubscriptionSearch
+    # Set the report options.
+    #
+    # # +per_page+: the page size.
+    # # +starts_at+: the report's starting date. Can't be older than 6 months.
+    # # +ends_at+: the report's ending date. Can't be greater than 30 days from the starting date.
+    #
+    attr_reader :options
+
+    # Set the errors from the report request.
+    attr_reader :errors
+
+    # Return the current page.
+    attr_reader :page
+
+    def initialize(path, options)
+      @path = path
+      @page = options.delete(:page) || 0
```

Comment:
Yes... The default page is 0. It's sound strange to me too, but this is the pattern the another Search used in **PagSeguro::Transaction**. https://github.com/pagseguro/ruby/blob/master/lib/pagseguro/transaction/search.rb#L17

This will allow the behaviour:
```ruby
# In this time the page's report is 0 unless you pass the page param
report = Subscription.search_by_date_interval(args)

while report.next_page?
  report.next_page! # In this time the page changes to 1 and the search report begin
  report.methods
  #... 
end

```